### PR TITLE
support for sunos5 in get_machine_id()

### PR DESCRIPTION
**Pull request by mtelka**
.. to get tests passed on illumos.

### CODE REVIEW

File: src/werkzeug/debug/__init__.py [hunk 1]
```diff
@@ -116,6 +116,14 @@ def _generate() -> str | bytes | None:
             except OSError:
                 pass
 
+        # On illumos and Solaris use the hostid(1) tool.
+        if sys.platform == "sunos5":
+            from subprocess import Popen, PIPE
+
+            hostid = Popen(["/usr/bin/hostid"], stdout=PIPE).communicate()[0]
+            if hostid != "":
```
Comment (ThiefMaster) [inline src/werkzeug/debug/__init__.py:124]:
```suggestion
            if hostid:
```

**Comment by ThiefMaster** [comment]
Is this an operating system still used in the wild nowadays? seems a bit esoteric, so not sure if we want explicit code for it, especially in a pure development util

**Comment by mtelka** [comment]
> Is this an operating system still used in the wild nowadays? seems a bit esoteric, so not sure if we want explicit code for it, especially in a pure development util

Of course, it is.  An illumos distro: https://www.openindiana.org/

**Comment by davidism** [comment]
Thanks, but I don't plan to merge this. It implies a level of support we can't provide. The debugger will still work without this (and note that the dev server and debugger should never be accessible externally anyway, so being able to generate a more random pin is the _last_ thing to worry about).

**Comment by mtelka** [comment]
Okay, np.  Your project, your rules. :-)

**Pull request closed without merging**